New Android embedding is only enabled on master

`hostonly_devicelab_tests` are failing _only in non-master branches_ on Cirrus with the following error.

```
[module_test] [STDOUT] stderr: /private/var/folders/sy/2x5qvs0n4lg18fry9jz4y21m0000gn/T/flutter_module_test.vOxl3v/hello_host_app/app/src/main/java/io/flutter/add2app/MainActivity.java:11: error: incompatible types: FlutterEngine cannot be converted to PluginRegistry
[module_test] [STDOUT] stderr:     GeneratedPluginRegistrant.registerWith(flutterEngine);
[module_test] [STDOUT] stderr:    
```

For example: https://cirrus-ci.com/task/4684340952301568

The following leads me to believe that the flutter tool is generating the wrong version of the plugin registrant:

```
[module_test] [STDOUT] stderr: C:\Users\ContainerAdministrator\AppData\Local\Temp\flutter_module_test.884c5db6-1601-11ea-aad5-8bd5e2fead93\hello\.android\Flutter\src\main\java\io\flutter\facade\Flutter.java:98: error: incompatible types: FlutterPluginRegistry cannot be converted to FlutterEngine
[module_test] [STDOUT] stderr:         GeneratedPluginRegistrant.registerWith(flutterView.getPluginRegistry());
[module_test] [STDOUT] stderr:                                                                             ^
```

Given that the project is referencing a `flutterView`, I think it should be using the old registrant version, but it appears to be using the new registrant version.

@blasten

--- a/packages/flutter_tools/lib/src/features.dart
+++ b/packages/flutter_tools/lib/src/features.dart
@@ -149,7 +149,19 @@ const Feature flutterAndroidEmbeddingV2Feature = Feature(
   name: 'flutter create generates projects using the Android embedding V2',
   environmentOverride: 'ENABLE_ANDROID_EMBEDDING_V2',
   configSetting: 'enable-android-embedding-v2',
+  beta: FeatureChannelSetting(
+    available: true,
+    enabledByDefault: true,
+  ),
+  dev: FeatureChannelSetting(
+    available: true,
+    enabledByDefault: true,
+  ),
   master: FeatureChannelSetting(
+    available: true,
+    enabledByDefault: true,
+  ),
+  stable: FeatureChannelSetting(
     available: true,
     enabledByDefault: true,
   ),

--- a/packages/flutter_tools/test/general.shard/features_test.dart
+++ b/packages/flutter_tools/test/general.shard/features_test.dart
@@ -440,6 +440,36 @@ void main() {
 
       expect(featureFlags.isWindowsEnabled, false);
     }));
+
+    group('isAndroidEmbeddingV2Enabled', () {
+      test('is enabled on beta', () => testbed.run(() {
+        when(mockFlutterVerion.channel).thenReturn('beta');
+        when<bool>(mockFlutterConfig.getValue('enable-android-embedding-v2') as bool).thenReturn(true);
+
+        expect(featureFlags.isAndroidEmbeddingV2Enabled, true);
+      }));
+
+      test('is enabled on dev', () => testbed.run(() {
+        when(mockFlutterVerion.channel).thenReturn('dev');
+        when<bool>(mockFlutterConfig.getValue('enable-android-embedding-v2') as bool).thenReturn(true);
+
+        expect(featureFlags.isAndroidEmbeddingV2Enabled, true);
+      }));
+
+      test('is enabled on master', () => testbed.run(() {
+        when(mockFlutterVerion.channel).thenReturn('master');
+        when<bool>(mockFlutterConfig.getValue('enable-android-embedding-v2') as bool).thenReturn(true);
+
+        expect(featureFlags.isAndroidEmbeddingV2Enabled, true);
+      }));
+
+      test('is enabled on stable', () => testbed.run(() {
+        when(mockFlutterVerion.channel).thenReturn('stable');
+        when<bool>(mockFlutterConfig.getValue('enable-android-embedding-v2') as bool).thenReturn(true);
+
+        expect(featureFlags.isAndroidEmbeddingV2Enabled, true);
+      }));
+    });
   });
 }
 

